subprojects/testing-base/src/main/java/org/gradle/api/internal/tasks/testing/logging/TestEventLogger.java

```diff
@@ -81,8 +94,35 @@ private void after(TestDescriptor descriptor, TestResult result) {
 
         if (shouldLogEvent(descriptor, event)) {
             String details = shouldLogExceptions(result) ? exceptionFormatter.format(descriptor, result.getExceptions()) : null;
+            logEventWithBuildOperation(descriptor, event, details);
+        }
+    }
+
+    private void logEventWithBuildOperation(TestDescriptor descriptor, TestLogEvent event, @Nullable String details) {
+        buildOperationExecutor.run(new TestLogEventBuildOperation(descriptor, event, details));
```

Comment:
We shouldn't use a build operation here. It would be much better to do this generally, but at the very least, we should reuse the stuff from the fix for `Exec` (a runnable that can temporarily associate a build operation id with the current thread.